third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -234,19 +238,27 @@ where
             ref_state: RefCheckState::new(),
         };
 
+        // Tracing and runtime checks (full or partial) are mutually exclusive because if we record
+        // the trace, the checks are done after execution via abstract interpretation during trace
+        // replay.
+        let paranoid_type_checks =
+            !trace_logger.is_enabled() && interpreter.vm_config.paranoid_type_checks;
```

Comment:
let's review the comments of paranoid_type_checks and optimize_trusted_code members in vm_config - maybe it could be helpful to mention that they may be deferred to the trace replay in the interpreter. Especially since ref checks aren't deferred. On that thought, maybe it makes sense to also assign
`let paranoid_ref_checks = interpreter.vm_config.paranoid_ref_checks;`

Reply:
Ref checks are not deferred yet because we also never benchmarked them and not running in production yet. Once @vineethk is back, we can port these to async as well, but need to support natives there first and see performance overhead + adjust check strategy: these will be much more heavy so we might have its own task for it even

Reply:
Added comments to vm config